antarest/login/utils.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024, RTE (https://www.rte-france.com)
+#
+# See AUTHORS.txt
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# SPDX-License-Identifier: MPL-2.0
+#
+# This file is part of the Antares project.
+
+import contextlib
+import typing as t
+from contextvars import ContextVar
+from typing import Optional
+
+from starlette.middleware.base import BaseHTTPMiddleware, RequestResponseEndpoint
+from starlette.requests import Request
+from starlette.responses import Response
+from starlette.types import ASGIApp
+from typing_extensions import override
+
+from antarest.core.jwt import JWTUser
+from antarest.fastapi_jwt_auth import AuthJWT
+from antarest.fastapi_jwt_auth.exceptions import AuthJWTException
+from antarest.login.auth import Auth
+
+_current_user: ContextVar[t.Optional[JWTUser]] = ContextVar("_current_user", default=None)
+
+
+class CurrentUserMiddleware(BaseHTTPMiddleware):
+    def __init__(self, app: ASGIApp, auth: Auth) -> None:
+        super().__init__(app)
+        self.auth = auth
+
+    @override
+    async def dispatch(self, request: Request, call_next: RequestResponseEndpoint) -> Response:
+        global _current_user
+        auth_jwt = AuthJWT(request)
+
+        try:
+            jwt_user = self.auth.get_current_user(auth_jwt)  # fail when no cookies provided
```

Comment:
I did not think about it too much before:
here we introduce a small technical debt because we perform 2x the authentication (I mean transforming the token into a user object), for the same request (once here, and another time when we add the jwt user in endpoints dependencies).

It blurs a little the responsibilities of who is actually doing the authentication, I think we should add a `TODO` comment here to mention this issue that we should try to resorb in the future.
Ideally the authentication should be done in one place, and the "result" (the user) re-used in the other places. I am not sure yet of the best way to do this, so we could keep it this way for this PR.